prime-router/src/main/kotlin/history/azure/ReportFileFunction.kt

```diff
@@ -0,0 +1,82 @@
+package gov.cdc.prime.router.history.azure
+
+import gov.cdc.prime.router.azure.WorkflowEngine
+import org.apache.logging.log4j.kotlin.Logging
+import java.time.OffsetDateTime
+import java.time.format.DateTimeParseException
+import java.util.UUID
+
+/**
+ * Submissions API
+ * Returns a list of Actions from `public.action`.
+ */
+abstract class ReportFileFunction(
+    internal val workflowEngine: WorkflowEngine = WorkflowEngine(),
+) : Logging {
+    data class Parameters(
+        val sort: String,
+        val sortColumn: String,
+        val cursor: OffsetDateTime?,
+        val endCursor: OffsetDateTime?,
+        val pageSize: Int,
+        val showFailed: Boolean
+    ) {
+        constructor(query: Map<String, String>) : this (
+            extractSortOrder(query),
+            extractSortCol(query),
+            extractCursor(query, "cursor"),
+            extractCursor(query, "endcursor"),
+            extractPageSize(query),
+            extractShowFailed(query)
+        )
+
+        companion object {
+            fun extractSortOrder(query: Map<String, String>): String {
+                return query.getOrDefault("sort", "DESC")
+            }
+
+            fun extractSortCol(query: Map<String, String>): String {
+                return query.getOrDefault("sortcol", "default")
+            }
+
+            fun extractCursor(query: Map<String, String>, name: String): OffsetDateTime? {
```

Comment:
Rename to extractDateTime or something?  With comment above not all dateTime values passed will be cursors.

Reply:
Makes sense. Updated